rust-k256/src/lib.rs

```diff
@@ -106,24 +105,20 @@ impl PlumeSignature<'_> {
         // don't forget to check `c` is `Output<Sha256>` in the #API
         let c = Output::<Sha256>::from_slice(self.c);
         // TODO should we allow `c` input greater than BaseField::MODULUS?
-        let c_scalar = &Scalar::from_uint_reduced(U256::from_be_byte_array(c.to_owned()));
+        let c_scalar = &Scalar::reduce_nonzero(U256::from_be_byte_array(c.to_owned()));
```

Comment:
Looks like `reduce_nonzero_bytes` will be more appropriate here, does it?

Also IIRC @Divide-By-0 mentioned that `c_scalar` indeed shouldn't be zero, and the update renders next check excessive (remove it, pls).

Reply:
I've looked into this one a bit deeper as @Divide-By-0 asked to make PRs merge easier, and I realize my comment isn't quite correct. I perceived `reduce_nonzero` as returning a non-zero `Scalar`, but actually it requires non-zero input. I believe the best option here is to use `NonZeroScalar`; or at least to check the value before the call.